gadget/install/install.go

```diff
@@ -796,15 +844,19 @@ func FactoryReset(model gadget.Model, gadgetRoot string, kernelSnapInfo *KernelS
 
 	// after we have created all partitions, build up the mapping of volumes
 	// to disk device traits and save it to disk for later usage
-	optsPerVol := map[string]*gadget.DiskVolumeValidationOptions{
-		// this assumes that the encrypted partitions above are always only on the
-		// system-boot volume, this assumption may change
-		bootVol.Name: {
-			ExpectedStructureEncryption: volCompatOps.ExpectedStructureEncryption,
-		},
+	optsPerVol := make(map[string]*gadget.DiskVolumeValidationOptions)
+	for name := range volumes {
+		if name == bootVol.Name {
```

Comment:
this is identical between Run and FactoryReset, wonder if it's useful to extract that as a helper maybe.

Reply:
right, this is a good observation, i actually simplified the construction of the code as contrary to what the loop might have suggested it's pretty static for now. this may change in the future, for now its too simple to reuse, and probably makes more sense to keep as-is now so its easier to change again